outlines/fsm/parsing.py

```diff
@@ -623,7 +619,7 @@ def match(self, text, pos, last_fsm_state_seq: Optional[Tuple[int, ...]] = None)
             text_part,
         )
 
-        state_seq = walk_fsm(
+        state_seq = walk_fsm(  # type: ignore  # noqa: F821
```

Comment:
I'm not sure if I should remove the whole `PartialScanner`, or workaround that?

Reply:
We need to keep everything that is related to grammars for now